app/src/main/java/org/simple/clinic/encounter/Encounter.kt

```diff
@@ -33,6 +36,24 @@ data class Encounter(
   interface RoomDao {
 
     @Insert(onConflict = REPLACE)
-    fun save(encounters: List<Encounter>)
+    fun save(encounters: List<Encounter>): Completable
+
+    @Query("UPDATE Encounter SET syncStatus = :newStatus WHERE syncStatus = :oldStatus")
+    fun updateSyncStatus(oldStatus: SyncStatus, newStatus: SyncStatus): Completable
```

Comment:
Curious, what is the rationale behind changing the `SyncStatus` for all entries. The criteria being a specific `SyncStatus`?

Reply:
NVM, this is enforced by the repository contract.